### ISSUE TITLE
Change Request: Expose Paging for All Languages

### ISSUE THREAD

**Issue by adrianhall** [Azure/azure-sdk#362]
## The Basics

* _Which design guideline is affected?_

Paging

* _Which target languages are affected?_

General Guidelines and all languages.

* Describe the change:

Instead of:

```
~ MustNot {#general-pagination-low-level-APIs}
expose low-level APIs to interact with the paged endpoint directly. These low-level APIs provide more fine-grained control over the pagination process while also providing users without access to (or understanding of) async iterators to use your library.
~
```

Change to

```
~ Must {#general-pagination-low-level-APIs}
expose low-level APIs to interact with the paged endpoint directly. These low-level APIs provide more fine-grained control over the pagination process while also providing users without access to (or understanding of) async iterators to use your library.
~
```

## Reasoning

In the case of ASP.NET applications (or really any RESTful API) that exposes a paged collection to the user, you want to pass through the paged operation to the backend service.  This is because you cannot guarantee the request for the next page comes in to the same backend.  

To correct this, you need to persist the continuation token you receive from the backend data store and return it to the client.  When the client re-submits (potentially to a different machine), it submits the continuation token from the data store, allowing the paging to pick up.

The alternative is for the ASP.NET application to re-read the complete collection, which is more expensive on the data store.  In many cases, this translates to "more money" being spent by the consumer for no real gain.

**Comment by Petermarcu** [comment]
Do you need to change the proposed wording to be different than the original? We also had a good discussion yesterday about the Cosmos API and exposing call cost that should be reconciled here.

**Comment by adrianhall** [comment]
I missed off the edit - turning MustNot into Must.

I'll invite chrande and kirill to the meeting.

**Comment by adrianhall** [comment]
Board decision:
- Defer the decision on this requirement until after the meeting about #365

**Comment by adrianhall** [comment]
Board decision:

* MUST including paging APIs when iterating over a collection
* Paging APIs must provide a continuation token as part of the response so that the iterator may continue, potentially on a different machine.
* Paging APIs must accept a continuation token (from a prior run) and a maximum count.

**Comment by adrianhall** [comment]
Merged in #384
